Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -254,84 +253,90 @@ function xASL_wrp_ResampleASL(x)
 				% Number of total volumes cannot be divided by blockSize
 				error(['Total number of volumes ' xASL_num2str(size(ASL_im,4)) ' cannot be composed of blocks of size ' xASL_num2str(blockSize)]);
 			else
-				PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
+				PWI4D = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
 				for iBlock = 1:blockSize
-					PWI(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
+					PWI4D(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions

```

Comment:
There seems to be an error here. You have averaged across repetitions. So this one is really PWI. And without the averages it is PWI4D.

Reply:
I see. So it should be:

PWI4D = xASL_quant_HadamardDecoding(......
PWI3D (:,:,:,iBlock) = xASL_stat_MeanNan(PWI4D(:,:,:, iBlock:....

?

But how do I then know which TE/PLD/labdur we have for each block?

Reply:
Yes. This fix helped. And then you went through the TEs. So the code now seems to be correct.

Reply:
OK then please resolve this conversation, otherwise there are far too many for me to check and re-read ;)